modules/aws-ssosync/iam.tf

```
@@ -0,0 +1,42 @@
+
+data "aws_iam_policy_document" "ssosync_lambda_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "ssosync_lambda_identity_center" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "identitystore:DeleteUser",
+      "identitystore:CreateGroup",
+      "identitystore:CreateGroupMembership",
+      "identitystore:ListGroups",
+      "identitystore:ListUsers",
+      "identitystore:ListGroupMemberships",
+      "identitystore:IsMemberInGroups",
+      "identitystore:GetGroupMembershipId",
+      "identitystore:DeleteGroupMembership",
+      "identitystore:DeleteGroup"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role" "default" {
+  count = var.enabled ? 1 : 0
```

```suggestion
  count = local.enabled ? 1 : 0
```
and define local.enabled is not already